File: app/src/main/java/com/example/contact/controller/Controller.java
package com.example.contact.controller;

import com.example.contact.model.ContactItem;

public final class Controller {

    public static  Controller controller=null;
    private ContactItem contactItem;

    /* singleton pattern */
    private Controller(){
        super();
    }

    public static Controller getController(){
        if(controller == null){
            controller = new Controller();

        }
        return controller;

    }

    public void nouveauContact(String nom, String prenom, String phone, String adresse, String email){

    }
}

File: app/src/main/java/com/example/contact/controller/Controller.kt
package com.example.contact.controller

import com.example.contact.model.ContactItem
import org.junit.runner.RunWith

class Controller  /* singleton pattern */
private constructor() {
    private val contactItem: ContactItem? = null
    fun nouveauContact(nom: String?, prenom: String?, phone: String?, adresse: String?, email: String?) {}

    companion object {
        var controller: Controller?=null
        fun Controller(): Controller? {

            if (controller == null) {
                controller = Controller()
            }
            return controller
        }
    }
}